Bio/SeqAnalysis/SeqDatabase.py

```diff
@@ -0,0 +1,277 @@
+import os
+import re
+from multiprocessing.pool import ThreadPool
+from os.path import isfile, join
+from time import time as timer
+from typing import Union, List, Tuple
+from urllib import request, error
+
+from Bio import Entrez
+from Bio import SeqIO
+
+
+class SeqDownloader:
+    """
+    SeqDownloader class is responsible for downloading fasta files for Uniprot/NCBI IDs
+    corresponding to given list or single ID ensuring their uniqueness.
+    """
+
+    def __init__(self, full_list: Union[str, List], directory: str):
+        """
+        In several steps, __init__ method is downloading and saving multiple fasta files in a short period of time:
+        1) Checks if the parameters are of the correct type, if not, raises exceptions
+        2) Checks if directories needed for proper file organization exists, if not, creates them
+        3) Prepares list of urls based on the list of IDs from which files will be obtained
+        4) Downloads and saves files using multiprocessing
+        5) Saves database in one fasta file ready for parse by SeqIO module. Also saves additional outputs which are
+           text files that contains entries not downloaded either because of being obsolete or due to http errors
+
+        :param full_list: Either List of IDs or single str.
+        :param directory: str given by user to organize files properly.
+
+        :raises: Exception when variables are not of the expected type.
+
+        """
+        if isinstance(full_list, str):
+            clear_ids = [full_list]
+        elif isinstance(full_list, List):
+            clear_ids = full_list
+        else:
+            raise Exception("Given variable is not either str or List.")
+        if isinstance(directory, str):
+            analysis_dir = directory
+        else:
+            raise Exception("Your directory name is not str.")
+
+        clear_ids = list(dict.fromkeys(clear_ids))
+
+        for list_element in clear_ids:
+            if isinstance(list_element, str):
+                pass
+            else:
+                raise Exception("At least one element of your IDs list is not str.")
+
+        print("Number of unique IDs: {0}".format(len(clear_ids)))
+        print("Name of the current batch of files: {0}".format(analysis_dir))
+
+        print("------------------------------------Directory check------------------------------------------")
```

Comment:
For all prints: instead of `print` you may want to use (https://docs.python.org/3/library/logging.html)[logging] library. It's not required, it's just good practice.